doc/man/man8/strlogd.8.man
'\" rtp
.\" vim: ft=nroff sw=4 noet nocin nosi com=b\:.\\\" fo+=tcqlorn tw=77
.\" =========================================================================
.\"
.\" @(#) doc/man/man8/strlogd.8.man
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2008-2015  Monavacon Limited <http://www.monavacon.com/>
.\" Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
.\" Copyright (c) 1997-2001  Brian F. G. Bidulock <bidulock@openss7.org>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to copy, distribute and/or modify this manual under
.\" the terms of the GNU Free Documentation License, Version 1.3 or any later
.\" version published by the Free Software Foundation; with no Invariant
.\" Sections, no Front-Cover Texts, and no Back-Cover Texts.  A copy of the
.\" license is included in the section entitled "GNU Free Documentation
.\" License".
.\"
.\" Permission to use, copy and distribute this manual without modification,
.\" for any purpose and without fee or royalty is hereby granted, provided
.\" that both the above copyright notice and this permission notice appears
.\" in all copies and that the name of OpenSS7 Corporation not be used in
.\" advertising or publicity pertaining to distribution of this documentation
.\" or its contents without specific, written prior permission.  OpenSS7
.\" Corporation makes no representation about the suitability of this manual
.\" for any purpose.  It is provided "as is" without express or implied
.\" warranty.
.\"
.\" Permission is granted to process this file through groff and print the
.\" results, provided the printed document carries a copying permission
.\" notice identical to this one except for the removal of this paragraph
.\" (this paragraph not being relevant to the printed manual).
.\"
.\" OPENSS7 CORPORATION DISCLAIMS ALL WARRANTIES WITH REGARD TO THIS MANUAL
.\" INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
.\" PARTICULAR PURPOSE, NON-INFRINGEMENT, OR TITLE; THAT THE CONTENTS OF THE
.\" DOCUMENT ARE SUITABLE FOR ANY PURPOSE, OR THAT THE IMPLEMENTATION OF SUCH
.\" CONTENTS WILL NOT INFRINGE ON ANY THIRD PARTY PATENTS, COPYRIGHTS,
.\" TRADEMARKS OR OTHER RIGHTS.  IN NO EVENT SHALL OPENSS7 CORPORATION BE
.\" LIABLE FOR ANY DIRECT, INDIRECT, SPECIAL OR CONSEQUENTIAL DAMAGES OR ANY
.\" DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
.\" IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING
.\" OUT OF OR IN CONNECTION WITH ANY USE OF THIS DOCUMENT OR THE PERFORMANCE
.\" OR IMPLEMENTATION OF THE CONTENTS THEREOF.
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this manual
.\" page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from the
.\" use of the information contained herein.  The author(s) may not have
.\" taken the same level of care in the production of this manual, which is
.\" licensed free of charge, as they might when working professionally.  The
.\" author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by the
.\" source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
.\" behalf of the U.S. Government ("Government"), the following provisions
.\" apply to you.  If the Software is supplied by the Department of Defense
.\" ("DoD"), it is classified as "Commercial Computer Software" under
.\" paragraph 252.227-7014 of the DoD Supplement to the Federal Acquisition
.\" Regulations ("DFARS") (or any successor regulations) and the Government
.\" is acquiring only the license rights granted herein (the license rights
.\" customarily provided to non-Government users).  If the Software is
.\" supplied to any unit or agency of the Government other than DoD, it is
.\" classified as "Restricted Computer Software" and the Government's rights
.\" in the Software are defined in paragraph 52.227-19 of the Federal
.\" Acquisition Regulations ("FAR") (or any successor regulations) or, in the
.\" cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the
.\" FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database openss7.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
search-truncate 40
search-ignore CGIQOSTU
.R2
.so openss7.macros
.\"
.\"
.TH STRLOGD 8 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ Administration"
.\"
.\"
.SH NAME
.B strlogd
\- \fISTREAMS\fR logger \fBsyslogd\fR(8) helper
.\"
.\"
.SH SYNOPSIS
.PP
.ad l
.nh
.HP \w'strlogd\ 'u
\fBstrlogd\fR [\fIoptions\fR]
[\fB-nqDv\fR]
[{\fB-d\fR|\fB--directory\fR[=]}\fIdirectory\fR]
[{\fB-p\fR|\fB--pidfile\fR[=]}[\fIpidfile\fR]]
[{\fB-l\fR|\fB--logdev\fR[=]}[\fIlogdevice\fR]]
[{\fB-D\fR|\fB--debug\fR[=]}[\fIlevel\fR]]
[{\fB-v\fR|\fB--verbose\fR[=]}[\fIlevel\fR]]
.PD 0
.HP
\fBstrlogd\fR {\fB-h\fR|\fB--help\fR}
.HP
\fBstrlogd\fR {\fB-V\fR|\fB--version\fR}
.HP
\fBstrlogd\fR {\fB-C\fR|\fB--copying\fR}
.PD
.ad b
.hy 6
.\"
.\"
.SH DESCRIPTION
.PP
The
.B strlogd
deamon provides some capabilities for the
.I STREAMS
logger not provided by
.BR syslogd (8).
The
.B strlogd
daemon receives console log messages from the
.I STREAMS
log driver
.RB ( log (4))
and issues them to
.BR syslog (3).
.BR syslogd (8)
is then responsible for processing the messages.
.PP
By default,
.B strlogd
logs all
.I STREAMS
console messages from all drivers and modules.
.PP
Messages that appear in the console log are intended to report exceptional
conditions that require the attention of a person who administers your system.
Those messages indicating the total failure a
.I STREAMS
driver or module will be logged with
.BR LOG_EMERG .
Those messages requiring the immediate attention of the administrator, and which should be mailed to
the system adminstrator must be mailed using the facilities of
.BR syslogd (8).
The
.BR strlogd
console logger will not mail these messages (although the error logger,
.BR strerr (8),
will).
The priority level has no meaning in the console log, but does have meaning if the message is also
sent to the trace logger.
.PP
Once initiated, the
.B strlogd
daemon continues to execute until terminated by the user.  Usually, the
.B strlogd
daemon is executed asynchronously.
.\"
.\"
.SH OPTIONS
.PP
.TP
\fB-d\fR, \fB--directory\fR \fIdirectory\fR
Specifies a directory to contain the messages.  Default is to search for
.RB \(lq /var/adm/streams \(rq
and then
.RB \(lq /var/log/streams .\(rq
.TP
\fB-n\fR, \fB--nodaemon\fR
Do not daemonize, run in the foreground, print output to
.I stdout
and error logs to
.IR stderr .
This option is most usefull for debugging
.BR strlogd .
.TP
\fB-b\fR, \fB--basename\fR \fIbasename\fR
Specify the log file basename.  Default is
.RB \(lq console \(rq .
.TP
\fB-o\fR, \fB--outfile\fR \fIoutfile\fR
Specifiy the log file output name.
When run as a daemon, the default is
.RI \(lq /dev/null \(rq
as no output is generated.
When run in the foreground, the default is ignored and all output is to
.IR stdout .
When specified, standard output is redirected to the specified file regardless of whether
.B strlogd
is being run in the foreground or as a daemon.
.TP
\fB-e\fR, \fB--errfile\fR \fIerrfile\fR
Specify the error file output name.
When run as a daemon, the default is
.RI \(lq /dev/null \(rq
and all errors are logged to
.BR syslogd (8).
When run in the foreground, the default is ignored and all errors are logged to
.IR stderr .
When specified, standard error is redirected to the specified file regardless of whether
.B strlogd
is being run in the foreground or as a daemon.
.TP
\fB-p\fR, \fB--pidfile\fR \fIpidfile\fR
Specify the pid file output name.  Default is
.RB \*(lq /var/run/strlogd.pid .\*(rq
When run in the foreground, the default is ignored.
.TP
\fB-l\fR, \fB--logdev\fR \fIlogdevice\fR
Specifies the log device.  Default is to search
.RB \(lq /dev/streams/strlog ,\(rq
.RB \(lq /dev/streams/log ,\(rq
.RB \(lq /dev/strlog \(rq
and
.RB \(lq /dev/log .\(rq
.TP
\fB-q\fR, \fB--quiet\fR
Suppresses normal output.
This option is equivalent to
.BR --verbose = 0 .
.TP
\fB-D\fR, \fB--debug\fR [\fIlevel\fR]
Increase or specify the debug output
.IR level .
When a debug
.I level
has been specified,
.B strlogd
will log to
.IR stdout ,
prints debug messages to
.IR stderr ,
and will not place itself in the background.
This option may be repeated.
.TP
\fB-v\fR, \fB--verbose\fR [\fIlevel\fR]
Increase or specify the verbosity
.I level
of the output.  This option may be repeated.
.TP
\fB-h\fR, \fB--help\fR
When this option is encountered,
.B strlogd
displays usage information on
.I stdout
and exits.
No further processing of options or arguments is performed and no actions are taken.
.TP
\fB-V\fR, \fB--version\fR
When this option is encountered,
.B strlogd
displays version information on
.I stdout
and exits.
No further processing of options or arguments is performed and no actions are taken.
.TP
\fB-C\fR, \fB--copying\fR
When this option is encountered,
.B strlogd
displays copying and copyright information on
.I stdout
and exits.
No further processing of options or arguments is performed and no actions are taken.
.\"
.\"
.SH ARGUMENTS
.PP
.B strlogd
takes no non-option arguments.
.\"
.\"
.SH DIAGNOSTICS
.PP
An exit status of zero
.RB ( 0 )
indicates that the command was successful; one
.RB ( 1 )
indicates that an error occurred and a diagnostic message is printed to
.IR stderr ;
two
.RB ( 2 )
indicates that the option or argument syntax was in error and a
diagnostic message is printed to
.IR stderr .
.PP
The
.B --quiet
option suppresses the printing of diagnostic messages to
.IR stderr .
.\"
.\"
.SH SIGNALS
.PP
.B syslogd
catches the
.B SIGHUP
signal
.RB "(see " signal (7)).
.TP
.B SIGHUP
When the
.B SIGHUP
signal is caught,
.B syslogd
will close the current log file and open a new (or reopen the existing) log
file.  This is for use with
.BR logrotate (8).
.TP
.B SIGTERM
When the
.B SIGTERM
signal is caught,
.B syslogd
flushes and closes all of its log files and then exits.
.\"
.\"
.SH NOTICES
.PP
Only one
.B strlogd
daemon or process at a time is permitted to open the
.I STREAMS
log driver as a console logger.
If a module or driver is generating a large number of console
messages, running the console logger causes a degradation in
.I STREAMS
performance.  If a large number of messages are generated in a short time, the
log driver may not be able to deliver some of the messages.  This situation is
indicated by gaps in the sequence numbering of the messages in the log files.
.PP
When run in the foreground,
.B strlogd
runs until terminated by the caller.
.PP
Running
.B strlogd
can impair
.I STREAMS
performance, particularly for those modules and drivers that are sending the
messages.
Also be aware that
.B strlogd
may not be able to handle a large number of messages.  If drivers and modules
return messages to
.B strlogd
too quickly, some may be lost.
.PP
Due to performance considerations, only one
.B strlogd
command is permitted to open the
.I STREAMS
.B log
driver at a time.
Running the
.B strlogd
command has the most impact on the timing of modules and drivers generating
the error messages that are sent to the
.B strlogd
process.  If error messages are generated faster than the
.B strlogd
process can handle them, some of the messages will be lost.  This last case
can be determined by examining the sequence numbers on the error messages
output.
.\"
.\"
.\".SH EXAMPLES
.\".PP
.\"
.\"
.SH DEVICES
.PP
.TP
.BR /dev/streams/strlog ", " /dev/streams/log ", " /dev/strlog " or " /dev/log
the
.I STREAMS
logging device.
See
.BR strlog (4).
.\"
.\"
.SH FILES
.PP
.TP
.B /usr/sbin/strlogd
contains the
.B strlogd
command.
.TP
.BR /var/log/streams " or " /var/adm/streams
directory containing the error log files.
.TP
.BI /var/log/streams/console. "mm-dd \fR or " /var/adm/streams/trace. mm-dd
console log file.
.TP
.BR /etc/init.d/strlogd " or " /etc/strlogd.sh
an SysV or Debian init script for starting the console logger.
.\"
.\"
.SH "SEE ALSO"
.PP
.BR log (4),
.BR strlog (4),
.BR strlog (9),
.BR strlog (1),
.BR strerr (8),
.BR strclean (8),
.BR syslogd (8),
.BR syslog (3).
.\"
.\"
.SH BUGS
.PP
.B strlogd
has no known bugs.
.\"
.\"
.SH COMPATIBILITY
.PP
.B strlogd
is compatible with implementations of
.BR syslogd
on
.IR "SVR 4.2" ,
.[
svr42cr
.]
and systems based on
.IR "SVR 4.2" ,
such as
.IR AIX \(rg,
.IR HP-UX \(rg,
.IR OSF/1 \(rg,
.IR Solaris \(rg,
.IR SUPER-UX \(rg,
.IR UnixWare \(rg
and
.IR UXP/V \(rg,
with the following compatibility considerations:
.IP \(em \w'\(em\(em'u
Options
.BR -n ", " -D ", " -b ", " -o ", " -e ", " -p " and " -l ,
and all long options, are specific to this
.I @PACKAGE_TITLE@
implementation of
.BR strlogd
and should not be used by portable applications.
.IP \(em
No other implementation documents the
.BR -q ", " -D ", " -v ", " -h ", " -V " or " -C
options.
They should not be used by portable applications.
.IP \(em
Options
.BR --help " and " --version
are provided for compatibility with GNU coding standards;
.BR --copying ,
OpenSS7 coding standards.
.IP \(em
.IR AIX \(rg,
.IR HP-UX \(rg,
.IR OSF/1 \(rg,
.IR Solaris \(rg,
.IR SUPER-UX \(rg
and
.IR UnixWare \(rg,
do not document this daemon directly, on these systems the functions performed by the
.B strlogd
daemon are described as being performed by the
.BR syslogd (8)
daemon.
.IP \(em
All other implementations document the log file directory as
.B /var/adm/streams
instead of
.BR /var/log/streams .
.B /var/log/streams
is used under
.B Linux
for compatibility with the
.BR "Linux File Standard" .
For compatibility,
.B strlogd
first checks if the directory
.B /var/adm/streams
exists, and then creates
.B /var/log/streams
if necessary, unless a directory is specified with the
.B -d
option.
.IP \(em
Under
.IR "SVR 4.2"
.[
svr42cr
.]
and
.IR UnixWare \(rg
.[
uw7doc
.]
the
.I STREAMS
log device is called
.B /dev/log
rather than
.BR /dev/strlog .
.B Linux
already has a
.RB non- STREAMS
based log mechanism (BSD-style) logging device named
.BR /dev/log .
.IP \(em
Under
.IR OSF/1 \(rg
.[
osfdoc
.]
the
.I STREAMS
log device is called
.B /dev/streams/log
instead of
.BR /dev/strlog .
.IP \(em
For compatibility with these operating environments,
.B strlogd
attempts to open
.BR /dev/streams/log ", " /dev/strlog " and then " /dev/log
if not specified with the
.B -l
option.
The call to
.B strlogd
will fail if there is no
.IR STREAMS -based
log device in this search sequence, or specified with the
.B -l
option.
.\"
.\"
.SH CONFORMANCE
.PP
.IR "SVR 4.2" ;
.[
svr42cr
.]
.IR AIX \(rg,
.IR HP-UX \(rg,
.IR OSF/1 \(rg,
.IR Solaris \(rg,
.IR SUPER-UX \(rg,
.IR UnixWare \(rg
and
.IR UXP/V \(rg
documentation.
.[
aixdoc
.]
.[
hpuxdoc
.]
.[
osfdoc
.]
.[
soldoc
.]
.[
suxdoc
.]
.[
uw7doc
.]
.[
uxpdoc
.]
.\"
.\"
.SH HISTORY
.PP
.B strlogd
is new for the
.I @PACKAGE_TITLE@
package.
.\"
.\"
.[
$LIST$
.]
.TI


NEWS
OpenSS7 -- history of user visible changes.  2015-01-15
$Id: NEWS,v 1.1.2.25 2011-05-10 13:45:30 brian Exp $
Copyright (c) 2008-2015  Monavacon Limited. <http://www.monavacon.com/>
Copyright (c) 2001-2008  OpenSS7 Corporation. <http://www.openss7.com/>
Copyright (c) 1997-2001  Brian Bidulock <bidulock@openss7.org>
See the end for copying conditions (for this file).

The sections that follow provide information on OpenSS7 releases of the
'OpenSS7' package.

Major changes for release openss7-1.1.7.20150115
------------------------------------------------

This is the seventh release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Source code release on GitHub.

Major changes for release openss7-1.1.7.20131209
------------------------------------------------

This is the seventh release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Build corrections for RHEL 6.4.

   - Scripts for country boundary, shoreline and rivers database
     preparation.

   - Database for ANSI T1.101 point code assignments.

   - Suppress extransous /proc/1/comm error messages when testing for
     systemd.

   - Handle PUIAS to Springdale rename for PUIAS distribution.

Major changes for release openss7-1.1.7.20131123
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Correction to ca-cert handling in RHEL 6 install scriptlet.

   - RHEL 6 build changes.

   - Set permissions correctly when applying kernel module patch.

   - Do not duplicate certs when including ca-cert.

   - Avoid conflicting shelll variable in RPM install scriplet.

   - Remove /proc/1/comm checks for systemd in module loader.

Major changes for release openss7-1.1.7.20130209
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Corrections to RPM spec file.

Major changes for release openss7-1.1.7.20130129
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Compensate for RHEL6 lack of c_rehash.

   - Do not create sysfs nodes: they do not work properly.

   - Use no foreground flag in strace.service file.

   - More error traces for strerr logger.

   - Remove static device names (the do not work either).

   - Changes to support weak kernel modules.

   - Clean up weak module support.

Major changes for release openss7-1.1.7.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Better Archlinux install script support.

Major changes for release openss7-1.1.6.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Remove use of kill_litter_super().

Major changes for release openss7-1.1.3.20130125
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Updates and corrections to weak kernel module builds.

   - Repo support and corrections for Archlinux packages.

Major changes for release openss7-1.1.3.20130123
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Additional support for Archlinux.

   - Add udev rules for strace and strerr.

   - Add system unit files for strace and strerr.

   - Add repo support for Archlinux packages.

   - Build Archlinux packages better.

Major changes for release openss7-1.1.3.20130121
------------------------------------------------

This is the sixth release of the 'OpenSS7' package.  This is a
production grade release.  All existing validation test suites run clean
on supported distributions, production kernels and architectures.

Major features since the last release are as follows:

   - Build support for SLES 11.

   - Handle certificates better.

   - Get configuration files in the right places for SLES 11.

   - Support in RPM spec file for certificates on SLES 11.

   - Create a new OpenSS7 certificate.

   - Repository support for SLES 11.

   - RPM spec file reworked for new RPM release.

   - Better certificate handling for RHEL.

   - Additional build support for RHEL 6 with certificates.


src/util/init/strlogd.sh
#!/bin/sh
#
# @(#) src/util/init/strlogd.sh
# Copyright (c) 2008-2015  Monavacon Limited <http://www.monavacon.com>
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# strlogd:	start and stop strlogd facility
# update-rc.d:	stop 20 0 1 6 .
# config:	/etc/default/strlogd
# processname:	strlogd
# pidfile:	/var/run/strlogd.pid
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  It is \
#		responsible for starting and stopping the STREAMS error \
#		logger.  The STREAMS error logger should only be run under \
#		exceptional circumstances and this init script not activated \
#		automatically.
#
# LSB init script conventions
#
### BEGIN INIT INFO
# Provides: strlogd
# Required-Start: streams
# Required-Stop: streams
# Default-Start: 
# Default-Stop: 0 1 2 3 4 5 6
# X-UnitedLinux-Default-Enabled: no
# Short-Description: start and stop strlogd
# License: GPL
# Description:	This STREAMS init script is part of Linux Fast-STREAMS.  It is
#	responsible for starting and stopping the STREAMS error logger.  The
#	STREAMS error logger should only be run under exceptional circumstances
#	and this init script is not activated in any run level by default.
### END INIT INFO

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='strlogd'
config="/etc/default/$name"
processname="$name"
pidfile="/var/run/$processname.pid"
execfile="/usr/sbin/$processname"
desc="the STREAMS error logger"

[ -x $execfile -o "$1" = "stop" ] || exit 5

# Specify defaults

STRLOGDOPTIONS=
STRLOGD_MAILUID=
STRLOGD_DIRECTORY="/var/log/streams"
STRLOGD_BASENAME="error"
STRLOGD_OUTFILE=
STRLOGD_ERRFILE=
STRLOGD_LOGDEVICE="/dev/streams/log"
STRLOGD_OPTIONS=

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

RETVAL=0

umask 077

case :$VERBOSE in
    :no|:NO|:false|:FALSE|:0|:)
	redir='>/dev/null 2>&1'
	;;
    *)
	redir=
	;;
esac

build_options() {
    # Build up the options string
    STRLOGD_OPTIONS=
    [ -n "$STRLOGDOPTIONS" ] && \
	STRLOGD_OPTIONS="${STRLOGD_OPTIONS:+ }${STRLOGDOPTIONS}"
    [ -n "$STRLOGD_MAILUID" ] && \
	STRLOGD_OPTIONS="${STRLOGD_OPTIONS:+ }-a ${STRLOGD_MAILUID}"
    [ -n "$STRLOGD_DIRECTORY" ] && \
	STRLOGD_OPTIONS="${STRLOGD_OPTIONS:+ }-d ${STRLOGD_DIRECTORY}"
    [ -n "$STRLOGD_BASENAME" ] && \
	STRLOGD_OPTIONS="${STRLOGD_OPTIONS:+ }-b ${STRLOGD_BASENAME}"
    [ -n "$STRLOGD_OUTFILE" ] && \
	STRLOGD_OPTIONS="${STRLOGD_OPTIONS:+ }-o ${STRLOGD_OUTFILE}"
    [ -n "$STRLOGD_ERRFILE" ] && \
	STRLOGD_OPTIONS="${STRLOGD_OPTIONS:+ }-e ${STRLOGD_ERRFILE}"
    [ -n "$STRLOGD_LOGDEVICE" ] && \
	STRLOGD_OPTIONS="${STRLOGD_OPTIONS:+ }-l ${STRLOGD_LOGDEVICE}"
}

start() {
    echo -n "Starting $desc: $name "
    build_options
    start-stop-daemon --start --quiet --pidfile $pidfile \
	--exec $execfile -- $STRACE_OPTIONS
    RETVAL=$?
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

stop() {
    if [ -r $pidfile ]; then
	echo -n "Stopping $desc: $name "
	start-stop-daemon --stop --quiet --retry=1 --oknodo --pidfile $pidfile \
	    --exec $execfile
	RETVAL=$?
    else
	RETVAL=0
    fi
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

reload() {
    echo -n "Reloading $desc: $name "
    start-stop-daemon --stop --quiet --signal=1 --pidfile $pidfile \
	--exec $execfile
    RETVAL=$?
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|reload|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|reload|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	reload || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) src/util/init/strlogd.sh
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2008-2015  Monavacon Limited <http://www.monavacon.com/>
# Copyright (c) 2001-2008  OpenSS7 Corporation <http://www.openss7.com/>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU General Public License as published by the Free Software
# Foundation; version 3 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
# details.
#
# You should have received a copy of the GNU General Public License along with
# this program.  If not, see <http://www.gnu.org/licenses/>, or write to the
# Free Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# =============================================================================
# vim: ft=sh sw=4 tw=80
